modules/core/src/main/java/org/apache/ignite/internal/processors/rest/handlers/log/GridLogCommandHandler.java

```diff
@@ -0,0 +1,149 @@
+    /**
+     * Default log file start line number *
+     */
+    private static final int DEFAULT_FROM = 0;
+    /**
+     * Default log file end line number*
+     */
+    private static final int DEFAULT_TO = 1;
+    /**
+     * @param ctx Context.
+     */
+    public GridLogCommandHandler(GridKernalContext ctx) {
+        super(ctx);
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public Collection<GridRestCommand> supportedCommands() {
+        return SUPPORTED_COMMANDS;
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public IgniteInternalFuture<GridRestResponse> handleAsync(GridRestRequest req) {
+        assert req != null;
+        assert SUPPORTED_COMMANDS.contains(req.command());
+        if (log.isDebugEnabled())
+            log.debug("Handling log REST request: " + req);
+        GridRestLogRequest req0 = (GridRestLogRequest) req;
+        int from = DEFAULT_FROM;
+        if (req0.from() != -1) {
+            from = req0.from();
+        }
+        int to = DEFAULT_TO;
+        if (req0.to() != -1) {
+            to = req0.to();
+        }
+        String path = DEFAULT_LOG_PATH;
+        Path filePath = getPath(req0, path);
+        if (from > to) {
+            return new GridFinishedFuture<>(new GridRestResponse(GridRestResponse.STATUS_FAILED,
+                    "Log file start from cannot be greater than to"));
+        }
+        switch (req.command()) {
+            case LOG:
+                try {
+                    List content = getContent(from, to, filePath);
+                    return new GridFinishedFuture<>(new GridRestResponse(content));
+                } catch (IgniteCheckedException e) {
+                    return new GridFinishedFuture<>(new GridRestResponse(GridRestResponse.STATUS_FAILED, e.getMessage()));
+                }
+        }
+
+        return new GridFinishedFuture<>();
+    }
+
+    private Path getPath(GridRestLogRequest req0, String path) {
+        Path filePath;
+        if (req0.path() != null) {
+            path = req0.path();
+            filePath = Paths.get(path);
+        } else {
+            filePath = Paths.get(ctx.config().getIgniteHome() + "/" + path);
+
+        }
+        return filePath;
+    }
+
+    private List getContent(int from, int to, Path filePath) throws IgniteCheckedException {
+        List content = new ArrayList<>();
+        try (BufferedReader reader = Files.newBufferedReader(filePath, Charset.defaultCharset())) {
+            String line = null;
+            int start = from;
+            int lineCounter = 0;
+            while ((line = reader.readLine()) != null
+                    && start < to) {
+                if (lineCounter >= from) {
+                    content.add(line);
+                    start++;
+                }
+                lineCounter++;
+            }
+        } catch (IOException e) {
+            throw new IgniteCheckedException("Unable to open file " + filePath.getFileName());
```

Comment:
The reason of IOException can be of other kind. I would thrown IgniteCheckedException(e) instead.